### PR TITLE
Data URI module

### PR DESCRIPTION
**Pull request by LorranSutter**

### CODE REVIEW

File: contracts/NFTs/VehicleId.sol [hunk 1]
```diff
@@ -80,6 +80,35 @@ contract VehicleId is Initializable, MultiPrivilege {
...
+    }
+
+    /**
+     * @notice Function to burn a token
+     * @dev To be called by DIMORegistry or a token owner
+     * DIMORegistry calls this function in burnVehicleSign function
+     * When a user calls it, burning is validated in the DIMORegistry
+     * @param tokenId Token Id to be burned
+     */
+    function burn(uint256 tokenId) public override {
```
Comment (elffjs) [inline contracts/NFTs/VehicleId.sol:103]: Just moving this around?

File: contracts/NFTs/VehicleId.sol [hunk 2]
```diff
@@ -80,6 +80,35 @@ contract VehicleId is Initializable, MultiPrivilege {
         trustedForwarders[addr] = trusted;
     }
 
+    /**
+     * @notice Gets the data URI associated to a token
+     * @param tokenId Token Id to be checked
+     * @return data Data URI
+     */
+    function getData(uint256 tokenId)
```
Comment (elffjs) [inline contracts/NFTs/VehicleId.sol:88]: The name seems odd to me. You're not getting the data!

`dataURI`? My guess is you're gonna want `dataUri` but note that it's `tokenURI` in the spec.

File: contracts/implementations/Nodes.sol [hunk 3]
```diff
@@ -35,4 +44,29 @@ contract Nodes {
...
+        returns (string memory data)
+    {
+        data = NodesStorage.getStorage().nodes[idProxyAddress][tokenId].info[
+            "Data"
+        ];
+
+        if (bytes(data).length == 0)
+            data = string(
+                abi.encodePacked(
+                    DataStorage.getStorage().dataUri,
```
Comment (elffjs) [inline contracts/implementations/Nodes.sol:67]: It seems unlikely to me that all node types would use the same base URL.

File: test/NFTs/VehicleId.test.ts [hunk 4]
```diff
@@ -554,6 +561,35 @@ describe('VehicleId', async function () {
     });
   });
 
+  describe('getDataURI', () => {
+    it('Should return the default data URI if not data is set in the token', async () => {
```
Comment (elffjs) [inline test/NFTs/VehicleId.test.ts:565]: Sick tests, as always.

File: utils/constants/Data.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export const BASE_DATA_URI = 'https://dimo.zone/vehicle/data/';
```
Comment (elffjs) [inline utils/constants/Data.ts:1]: One interesting thing is that our URLs right now look like `/vehicle/6/status`.

My guess is we'll just have to adjust.

**Review by elffjs — CHANGES_REQUESTED** [review]

File: utils/constants/Data.ts (hunk 5 above)
Comment (LorranSutter) [reply utils/constants/Data.ts:1]: Does not hurt change it in the code. What option you think is the easiest?

**Review by elffjs — APPROVED** [review]

File: contracts/NFTs/VehicleId.sol [hunk 6]
```diff
@@ -80,6 +80,35 @@ contract VehicleId is Initializable, MultiPrivilege {
         trustedForwarders[addr] = trusted;
     }
 
+    /**
+     * @notice Gets the data URI associated to a token
+     * @param tokenId Token Id to be checked
+     * @return data Data URI
+     */
+    function getDataURI(uint256 tokenId)
```
[the remark this answers is not in the document]
Comment (elffjs) [reply contracts/NFTs/VehicleId.sol:88]: I see you have decided on `getDataURI`.

**Pull request merged**